src/Storage/RedisStorage.php

```diff
@@ -20,28 +20,39 @@ class RedisStorage implements PersistenStorageInterface
      */
     private $redis;
 
-    public function __construct($redis)
+    /**
+     * @var string
+     */
+    private $namespace;
+
+    public function __construct($redis, ?string $namespace = null)
     {
         if (!$redis instanceof \Redis && !$redis instanceof \RedisArray && !$redis instanceof \RedisCluster && !$redis instanceof \Predis\Client) {
             throw new \InvalidArgumentException(sprintf('%s() expects parameter 1 to be Redis, RedisArray, RedisCluster or Predis\Client, %s given', __METHOD__, is_object($redis) ? get_class($redis) : gettype($redis)));
         }
 
         $this->redis = $redis;
+        $this->namespace = $namespace;
     }
 
     /**
      * {@inheritdoc}
      */
     public function get(string $key)
     {
-        return $this->redis->get($key);
+        return $this->redis->get($this->generateKey($key));
     }
 
     /**
      * {@inheritdoc}
      */
     public function set(string $key, bool $value)
     {
-        return $this->redis->set($key, $value);
+        return $this->redis->set($this->generateKey($key), $value);
+    }
+
+    private function generateKey(string $key): string
+    {
+        return !empty($this->namespace) ? sprintf('%s:%s', $this->namespace, $key) : $key;
```

Comment:
Can you please remove the `empty` check here, and just check it inline? And also use string interpolation for the values
E.G
```php
return $this->namespace ? "{$this->namespace}:$key" : $key;
```

Reply:
Much better solution. I will make the chang

Reply:
I pushed that change. Thanks for the suggestion